code/src/components/Survey.js

```diff
@@ -0,0 +1,207 @@
+  const onCheckboxGroupToggle =(checkboxValue) => {
+    if (checkboxGroup.includes(checkboxValue)) {
+      // delete element from array by using filter method
+      setCheckboxGroup(checkboxGroup.filter(item => item !== checkboxValue))
+    } else {
+      // add element to array. Spread operator (...) will copy all the values from the previous heckboxGroup array 
+      setCheckboxGroup([checkboxValue, ...checkboxGroup])
+    }
+  }
+
+
+  //RENDERING
+
+  return (
+    <main className='main-container'>
+      {/*This will show the the questions if the form is not submitted*/}
+      {!showSummary ? (
+      <form className='form' onSubmit={handleSubmit}>
+
+        {/* Survey starts here  */}
+        {question === 0 && (
+          <section className='start-page'>
+            <h1 className='header'>READING HABITS SURVEY</h1>
+            <p className='small-text'>Time to complete: 3 minutes</p>
+            <div className="start-button-container">
+              <Button onChangeDirection={onNextQuestionChange} textDisplay={'Start the survey'} className={'start-button'} />
+            </div>
+          </section>
+        )}
+
+        {/* First question */}
+        {question === 1 && (
+          <section className='question-container'>
+            <h2 tabindex='0' className='question'>How much do you enjoy reading?</h2>
+            <div className="radio-buttons">
+              {
+                enjoyReadingArray.map(item =>
+                  <InputRadioButton 
+                    item={item} 
+                    onEnjoyReadingChange={onEnjoyReadingChange} 
+                  />
+                )
+              }
+            </div>
+            <div className="button-container">
+              <Button onChangeDirection={onPreviousQuestionChange} textDisplay={'Go Back'} className={'button'} />
+              <Button onChangeDirection={onNextQuestionChange} textDisplay={'Continue'} className={'button'} />
+            </div>
+          </section>
+        )}
+
+        {/* Second question */}
+        {question === 2 && (
+          <section className='question-container'>
+            <InputDropDown 
+              question={'Which format do you prefer reading the most?'}
+              dropDownvalue={readingFormat}
+              onChangeFunction={onReadingFormatChange}
+            />
+            <div className="button-container">
+              <Button onChangeDirection={onPreviousQuestionChange} textDisplay={'Go Back'} className={'button'} />
+              <Button onChangeDirection={onNextQuestionChange} textDisplay={'Continue'} className={'button'} />
+            </div>
+          </section>
+        )}
+
+        {/* Third question */}
+        {question === 3 && (
+          <section className='question-container'>
+            <InputText 
+              question={'What is your favorite book?'}
+              inputTextValue={book}
+              onChangeFunction={onBookChange}
```

Comment:
When answering to this on your website with my favorite book (the alchemist) I didn't get the following questions, but received my summary directly. I then tried different books like (Harry Potter - not my favourite!🙃) and that also didn't work. So I first thought you had a problem with those books, but then I thought it might be because it was two words, but also writing them in one didn't work. So probably here is something limited to 5 letters (I think, cause with 6 letters it doesn't work anymore) - maybe something to look into.. I couldn't find anything suspicious🕵🏼‍♀️  🧐  🙃